crates/networks/src/lib.rs

```diff
@@ -98,6 +104,10 @@ impl Networks {
     }
 
     pub async fn update_network(&mut self, old_name: &str, network: Network) -> Result<()> {
```

Comment:
I think we should have a separate struct here, which is the network without chain ID
strictly speaking it's not part of the intended arguments.

and since this is currently disabled only on the form level, it's still technically possible to change

that said, I'm ok with opening a separate issue to deal with this later

Reply:
I'll make those changes together with the `internal_id` since I'll be touching that part of the code either way